src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -245,6 +246,13 @@ public async Task ProcessInputAsync(Message message, Flow flow, IContext message
                                     // Store the previous state
                                     await _stateManager.SetPreviousStateIdAsync(context, previousStateId, linkedCts.Token);
 
+                                    // Only execute the ProcessAfterStateActionsAsync when the user current state changed after ProcessOutputsAsync
+                                    if (previousState.Id != state?.Id)
```

Comment:
why not use "previousStateId" variable?

Reply:
Because we will need the State object (called Previus) to execute the AfterStateChanged actions and the previousStateId has already been changed earlier in the code to the current state id